spec/api/datastore/maat/application_ready_spec.rb

```diff
@@ -5,31 +5,53 @@
     let(:api_request) { get "/api/maat/applications/#{application_usn}" }
 
     let(:application) do
-      CrimeApplication.new(
+      CrimeApplication.create(
         application: JSON.parse(LaaCrimeSchemas.fixture(1.0).read),
         review_status: :ready_for_assessment,
+        id: SecureRandom.uuid,
+        submitted_at: 1.day.ago
       )
     end
 
     let(:application_usn) { application.application['reference'] }
-    let(:records) { JSON.parse(response.body).fetch('records') }
+    let(:maat_application) { JSON.parse(response.body) }
 
     context 'with a ready for assessment application' do
       before do
-        allow(CrimeApplication).to receive(:find_by)
-          .with(reference: application_usn, review_status: :ready_for_assessment)
-          .and_return(application)
-
         api_request
       end
 
       it 'returns http status 200' do
         expect(response).to have_http_status(:success)
       end
 
-      it 'returns the reference details' do
-        expect(records['reference']).to match(application.application['reference'])
-        expect(records['applicant']).to match(application.application['client_details']['applicant'])
+      it 'returns valid maat application details' do
+        expect(
+          LaaCrimeSchemas::Validator.new(response.body, version: 1.0, schema_name: 'maat_application')
+        ).to be_valid
+      end
+
+      it 'returns the required application details for maat integration' do
+        expect(maat_application['reference']).to match(application.application['reference'])
```

Comment:
I think it may be clearer to have a hash that we compare against rather than lots of expects for each field. I think there may be a matcher for this sort of thing already in rspec like you have `match_array` etc.

Reply:
The best matcher I could find to use was include but I wasn't sure it check the value of each key matches was is expected as it gets quite nested..  there should really be a match_array equivalent for hashes 🤔

I think it's clearer not but let me know if you had a different approach in mind